File: Homework/Quang_Minh/Lesson1/LinkedListCycle.java
public class LinkedListCycle {
    public boolean hasCycle(ListNode head) {
        ListNode fastPointer = head;
        ListNode slowPointer = head;
        while (true) {
            if (slowPointer == null || fastPointer == null || fastPointer.next == null) break;
            fastPointer = fastPointer.next;
            if (slowPointer == fastPointer) {
                return true;
            }
            fastPointer = fastPointer.next;
            if (slowPointer == fastPointer) {
                return true;
            }
            slowPointer = slowPointer.next;
        }
        return false;
    }
}

File: Homework/Khang_Nguyen/linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
 var hasCycle = function(head) {
    if (head === null) return false;
    let slow = head
    let fast = head

    while (true) {
      if (fast.visited) return true;
      fast.visited = true
      fast = fast.next
      if (fast === null) return false;
      if (fast.visited) return true;
      fast.visited = true
      fast = fast.next
      if (fast === null) return false;

      slow = slow.next
    } 
};